src/index.js

```diff
@@ -122,83 +152,119 @@ class ExtractTextPlugin {
 
   apply(compiler) {
     const options = this.options;
+
     compiler.plugin('this-compilation', (compilation) => {
       const extractCompilation = new ExtractTextPluginCompilation();
-      compilation.plugin('normal-module-loader', (loaderContext, module) => {
-        loaderContext[NS] = (content, opt) => {
-          if (options.disable) {
-            return false;
-          }
-          if (!Array.isArray(content) && content != null) {
-            throw new Error(
-              `Exported value was not extracted as an array: ${JSON.stringify(
-                content
-              )}`
-            );
-          }
-          module[NS] = {
-            content,
-            options: opt || {},
+
+      compilation.hooks.normalModuleLoader.tap(
+        'normal-module-loader',
```

Comment:
The tap name should be the name of the caller. Here it's `ExtractTextPlugin`

Reply:
```js
class ExtractTextPlugin {
   constructor() {
     this.plugin = { name: 'ExtractTextPlugin' };
     ...
   }

   apply(compiler) {
     const { plugin } = this;
     const { someHook } = compiler.hooks;
     
     someHook.tap(plugin, () => {});
   }
}
```